pkg/backend/httpstate/state.go

```diff
@@ -259,6 +276,12 @@ func (b *cloudBackend) newUpdate(ctx context.Context, stackRef backend.StackRefe
 		return nil, err
 	}
 
+	// We will reuse the snapshot's secrets manager when possible to ensure that secrets are not re-encrypted on each
+	// update.
+	if secrets.AreCompatible(sm, target.Snapshot.SecretsManager) {
+		sm = target.Snapshot.SecretsManager
```

Comment:
Should we fail if the secrets managers aren't compatible? e.g. the existing snapshot is using a Service or passphrase-based secrets manager, and instead we are passed a different type of secrets manager? If that were the case, wouldn't we know ahead of time that all requests to decrypt data would fail?